src/internal_modules/roc_core/target_openssl/roc_core/secure_random.cpp

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2025 Roc Streaming authors
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#include "secure_random.h"
+
+#include "roc_core/log.h"
+
+#include <openssl/err.h>
+#include <openssl/rand.h>
+
+namespace roc {
+namespace core {
+
+bool secure_random(void* buf, size_t bufsz) {
+    // RAND_priv_bytes() is not needed right now as we do not use these random numbers
+    // privately. See also https://docs.openssl.org/3.0/man7/RAND/
+
+    int ok = RAND_bytes((unsigned char*)buf, (int)bufsz);
+
+    if (ok != 1) {
+        unsigned long err = ERR_get_error();
+        const size_t err_str_len = 256; // minimum buf length is 256
+        char err_str[err_str_len];
+        ERR_error_string_n(err, err_str, err_str_len);
+        roc_log(LogError, "OpenSSL: failed to generate random number: %s", err_str);
```

Comment:
Nit: OpenSSL -> secure random (name of component which produces log message)

Reply:
module name is always written by `roc_log` itself, or I don't get something?
```
#define roc_log(level, ...)                                                              \
    do {                                                                                 \
        ::roc::core::Logger& logger = ::roc::core::Logger::instance();                   \
        if ((level) <= logger.get_level()) {                                             \
            logger.writef((level), ROC_STRINGIZE(ROC_MODULE), __FILE__, __LINE__,        \
                          __VA_ARGS__);                                                  \
        }                                                                                \
    } while (0)
```